Mathlib/MeasureTheory/Measure/AddContent.lean

```diff
@@ -0,0 +1,185 @@
+* `MeasureTheory.AaddContent_union'`: if `s, t ∈ C` are disjoint and `s ∪ t ∈ C`,
+  then `m (s ∪ t) = m s + m t`.
+  If `C` is a set ring (`IsSetRing`), then `addContent_union` gives the same conclusion without the
+  hypothesis `s ∪ t ∈ C` (since it is a consequence of `IsSetRing C`).
+
+If `C` is a set ring (`MeasureTheory.IsSetRing C`), we have, for `s, t ∈ C`,
+
+* `MeasureTheory.addContent_union_le`: `m (s ∪ t) ≤ m s + m t`
+* `MeasureTheory.addContent_le_diff`: `m s - m t ≤ m (s \ t)`
+
+-/
+
+open Set Finset
+
+open scoped ENNReal BigOperators
+
+namespace MeasureTheory
+
+variable {α : Type*} {C : Set (Set α)} {s t : Set α} {I : Finset (Set α)}
+
+/-- An additive content is a set function with value 0 at the empty set which is finitely additive
+on a given set of sets. -/
+structure AddContent (C : Set (Set α)) where
+  /-- The value of the content on a set. -/
+  toFun : Set α → ℝ≥0∞
+  empty' : toFun ∅ = 0
+  sUnion' (I : Finset (Set α)) (_h_ss : ↑I ⊆ C)
+      (_h_dis : PairwiseDisjoint (I : Set (Set α)) id) (_h_mem : ⋃₀ ↑I ∈ C) :
+    toFun (⋃₀ I) = ∑ u in I, toFun u
+
+instance : Inhabited (AddContent C) :=
+  ⟨{toFun := fun _ => 0
+    empty' := by simp
+    sUnion' := by simp }⟩
+
+instance : FunLike (AddContent C) (Set α) (fun _ ↦ ℝ≥0∞) where
+  coe := fun m s ↦ m.toFun s
+  coe_injective' := by
+    intro m m' h
+    cases m
+    cases m'
+    congr
+
+variable {m m' : AddContent C}
+
+@[ext] protected lemma AddContent.ext (h : ∀ s, m s = m' s) : m = m' :=
+  FunLike.ext _ _ h
+
+protected lemma AddContent.ext_iff (m m' : AddContent C) : m = m' ↔ ∀ s, m s = m' s :=
+  FunLike.ext_iff
+
+@[simp] lemma addContent_empty : m ∅ = 0 := m.empty'
+
+lemma addContent_sUnion (h_ss : ↑I ⊆ C)
+    (h_dis : PairwiseDisjoint (I : Set (Set α)) id) (h_mem : ⋃₀ ↑I ∈ C) :
+    m (⋃₀ I) = ∑ u in I, m u :=
+  m.sUnion' I h_ss h_dis h_mem
+
+lemma addContent_union' (hs : s ∈ C) (ht : t ∈ C) (hst : s ∪ t ∈ C)(h_dis : Disjoint s t) :
+    m (s ∪ t) = m s + m t := by
+  by_cases hs_empty : s = ∅
+  · simp only [hs_empty, Set.empty_union, addContent_empty, zero_add]
+  classical
+  have h := addContent_sUnion (m := m) (I := {s, t}) ?_ ?_ ?_
+  rotate_left
+  · simp only [coe_pair, Set.insert_subset_iff, hs, ht, Set.singleton_subset_iff, and_self_iff]
+  · simp only [coe_pair, Set.pairwiseDisjoint_insert, pairwiseDisjoint_singleton,
+      mem_singleton_iff, Ne.def, id.def, forall_eq, true_and_iff]
+    exact fun _ => h_dis
+  · simp only [coe_pair, sUnion_insert, sUnion_singleton]
+    exact hst
+  convert h
+  · simp only [coe_pair, sUnion_insert, sUnion_singleton]
+  · rw [sum_insert, sum_singleton]
+    simp only [Finset.mem_singleton]
+    refine fun hs_eq_t => hs_empty ?_
+    rw [← hs_eq_t] at h_dis
+    exact Disjoint.eq_bot_of_self h_dis
+
+section IsSetSemiring
+
+lemma addContent_eq_add_diffFinset₀_of_subset (hC : IsSetSemiring C)
+    (hs : s ∈ C) (hI : ↑I ⊆ C) (hI_ss : ⋃₀ ↑I ⊆ s)
```

Comment:
Thinking about it more, shouldn't this be spelled 
```suggestion
    (hs : s ∈ C) (hI : ↑I ⊆ C) (hI_ss : ∀ t ∈ I, t ⊆ s)
```
and elsewhere too?

Reply:
Indeed the simp lemma [Set.sUnion_subset_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Lattice.html#Set.sUnion_subset_iff) says your suggestion should be preferred.